app/src/main/java/com/nervousfish/nervousfish/activities/VisualVerificationActivity.java

```diff
@@ -74,12 +75,13 @@ private void nextActivity() {
         LOGGER.info("Done tapping the VisualVerification");
 
         try {
-            final Profile myProfile = this.serviceLocator.getDatabase().getProfiles().get(0);
+            final Profile profile = this.serviceLocator.getDatabase().getProfiles().get(0);
+            final KeyPair keyPair = profile.getKeyPairs().get(0);
 
-            LOGGER.info("Sending my profile with name: " + myProfile.getName() + ", public key: "
-                    + myProfile.getPublicKey().toString());
+            LOGGER.info("Sending my profile with name: " + profile.getName() + ", public key: "
+                    + keyPair.getPublicKey().toString());
 
-            final Contact myProfileAsContact = new Contact(myProfile.getName(), new SimpleKey("simplekey", "73890ien"));
+            final Contact myProfileAsContact = new Contact(profile.getName(), new Ed25519Key("simplekey", "73890ien"));
```

Comment:
Will change it, although this is not the code that should be there (it should sent the actual key of the user)